core/integration/git_test.go

```diff
@@ -170,12 +148,23 @@ sleep infinity
 		}
 	}()
 
-	entries, err := c.Git("ssh://root@127.0.0.1:3486/root/repo").
+	sshPort := 2222
+	sshSvc := hostKeyGen.
+		WithMountedFile("/root/start.sh", setupScript).
+		WithExposedPort(sshPort).
+		WithExec([]string{"sh", "/root/start.sh"})
+
+	sshHost, err := sshSvc.Hostname(ctx)
+	require.NoError(t, err)
+
+	repoURL := fmt.Sprintf("ssh://root@%s:%d/root/repo", sshHost, sshPort)
+	entries, err := c.Git(repoURL, dagger.GitOpts{ExperimentalServiceHost: sshSvc}).
```

Comment:
So this means that all the service hostnames of containers spun up by buildkit are also available to the buildkit process itself? AFAIK [buildkit runs git commands uncontainerized](https://github.com/sipsma/buildkit/blob/a8e8d2a0afad5b0c35c8605e864d666de407e285/source/git/gitsource.go#L660-L660) so I think that would have to be the case, but I could be missing something.

Totally fine if so I think, just want to double-check my understanding. I guess it would rule out supporting service aliases for these dependencies in the future but I can't really imagine why that'd be a problem right now.

Reply:
Yep! And yeah, if we wanted to support service aliases we'd probably have to resort to URL munging. That goes for HTTP, Git, and Container.From.

This works because we now run the engine with flags to configure its local `dnsmasq` in `/etc/resolv.conf`:

https://github.com/dagger/dagger/pull/4505/files#diff-96a7f5c352cd3d2d1ee3c7c55ed95e668c6edb28a4fa337e6c60270471bdab61

(`--dns 10.87.0.1 --dns-search dns.dagger` plus the original config from the host's `resolv.conf`)

This config is then inherited by containers we run, and for service aliases we modify the OCI bundle's `/etc/hosts` prior to running the shim.